core/src/main/java/com/datastrato/gravitino/utils/IsolatedClassLoader.java

```diff
@@ -225,7 +225,15 @@ private boolean isSharedClass(String name) {
    * @return true if the class is a catalog class, false otherwise.
    */
   private boolean isCatalogClass(String name) {
-    return name.startsWith("com.datastrato.gravitino.catalog");
+    return name.startsWith("com.datastrato.gravitino.catalog")
```

Comment:
Should we still include this line?

Reply:
Retaining it can act as a fail fast?

Reply:
So the semantics are changed, we want `catalog` package in `core` to be shared, right? With you change, it means not shared.

Reply:
> So the semantics are changed, we want catalog package in core to be shared, right?

yes

> With you change, it means not shared.

no

For example, class `com.datastrato.gravitino.catalog.CapabilityHelpers` in core will return `false` in this method(`isCatalogClass`) then result in `true` in [isSharedClass](https://github.com/datastrato/gravitino/pull/3300/files#diff-477e4b46c2b99dda8a31f82ce397250c1d418609d9936a833cfd8e56db9f9b8eR217), so it is shared class.

Reply:
> Retaining it can act as a fail fast?

It can't persuade me. What if the `name` is indeed a shard class, the method will add an extra and useless judge condition.

Reply:
I see, thanks.

Reply:
> > Retaining it can act as a fail fast?
> 
> It can't persuade me. What if the `name` is indeed a shard class, the method will add an extra and useless judge condition.

if the `name` is indeed a shared class, it needs 1 judge. Without this judge, it needs additional 8 times judge

Reply:
Why do change this `name.startsWith("com.datastrato.gravitino.")` to `name.startsWith("com.datastrato.gravitino.catalog")`?

Reply:
> Why do change this `name.startsWith("com.datastrato.gravitino.")` to `name.startsWith("com.datastrato.gravitino.catalog")`?

Almost all classes start with `com.datastrato.gravitino`, so judging `name.startsWith("com.datastrato.gravitino.")` is meaningless.

Reply:
> judge

OK, let me clarify it . 
- If the name is `com.datastrato.gravitino.catalog.CatalogMananger`,  then it should check 9 times at most. Removing it can only reduce it to 8 at most. 
- If the name is `com.datastrato.gravitino.catalog.hive.xxx`,  then it should check 9 times at most. Removing it can only reduce it to 8 at most.
- If the name is `com.datastrato.gravitino.xxxx(NOT catalog)`, then only needs 1, Removing it may take 8 times at most.

Sounds we'd better keep it.